test/e2e/e2e_toomanyrestarts_test.go

```diff
@@ -30,24 +30,27 @@ import (
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/events"
 	"k8s.io/utils/pointer"
-	deschedulerapi "sigs.k8s.io/descheduler/pkg/api"
+
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
 	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
 	eutils "sigs.k8s.io/descheduler/pkg/descheduler/evictions/utils"
-	"sigs.k8s.io/descheduler/pkg/descheduler/strategies"
+	"sigs.k8s.io/descheduler/pkg/framework"
+	frameworkfake "sigs.k8s.io/descheduler/pkg/framework/fake"
+	"sigs.k8s.io/descheduler/pkg/framework/plugins/removepodshavingtoomanyrestarts"
 )
 
 func TestTooManyRestarts(t *testing.T) {
 	ctx := context.Background()
 
-	clientSet, _, _, getPodsAssignedToNode, stopCh := initializeClient(t)
+	clientSet, sharedInformerFactory, _, getPodsAssignedToNode, stopCh := initializeClient(t)
 	defer close(stopCh)
 
 	nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		t.Errorf("Error listing node with %v", err)
 	}
 
-	nodes, workerNodes := splitNodesAndWorkerNodes(nodeList.Items)
+	nodes, _ := splitNodesAndWorkerNodes(nodeList.Items)
```

Comment:
It is still important to run the plugin over worker nodes only. Some of the control plane pod containers may get restarted too many times and evicted by the descheduler. Changing the assumption about the master nodes being protected against eviction and making the debugging step more complicated.